FreeTubeApp/FreeTube
# Import from YouTube history using JSON export

# Import from YouTube history using JSON export

## Pull Request Type

## Related issue
closes #1623

## Description
Imports the history of YouTube watched videos from Google Takeout when using the JSON format.

## Screenshots 
Added a button to import from YouTube History

> Before
> <img width="1018" alt="image" src="https://user-images.githubusercontent.com/326831/207443270-345f7c56-53ae-4239-9736-6027ece222a1.png">

> After
> <img width="1021" alt="image" src="https://user-images.githubusercontent.com/326831/207443337-5295a33c-172c-4ed1-95b2-a9ecabf9d171.png">

## Testing 

1. Go to YouTube
2. Click on your profile picture
3. Click on "Your data in YouTube" ![image](https://user-images.githubusercontent.com/326831/207443988-80cd9346-b8a3-4502-90e4-1dbe772a0b9f.png)
4. Click on "More" ![image](https://user-images.githubusercontent.com/326831/207444126-fa3e0b29-53d9-4e3e-832f-5a0b4ba8f5b6.png)
5.  Click on "Download YouTube Data"  ![image](https://user-images.githubusercontent.com/326831/207444219-edfc0ee4-ef1b-4b53-9037-dc5d7c1e8e4c.png)
6. Click on "All YouTube data included" ![image](https://user-images.githubusercontent.com/326831/207444409-69ce79e4-7e9d-4fe5-9ae3-9067997f4397.png)
7. Ensure that you have "history" option checked ![image](https://user-images.githubusercontent.com/326831/207444527-71b90cca-0be3-46f0-911a-af07e3b55f25.png)
8. Click on OK
9. Click on "HTML format" ![image](https://user-images.githubusercontent.com/326831/207444409-69ce79e4-7e9d-4fe5-9ae3-9067997f4397.png)
10. Change the type of the history to JSON ![image](https://user-images.githubusercontent.com/326831/207444822-0af72ad1-24b1-4c11-8ff6-232cfb5826a0.png)
11. Click OK
12. Click on "Next Step"
13. Choose your option to transfer and lick "Create export"
14. Wait the export to finish, download the package and unzip it
15. Open FreeTube
16. Go to "Settings"
17. Click on "Data Settings"
18. Click on "Import YouTube History"
19. Select the JSON file from the Google Takeout export
20. Enjoy your imported history

## Desktop
- **OS:** macOS
- **OS Version:** Ventura 13.0.1 (22A400)
- **FreeTube version:** v0.18.0 Beta (from development branch)

## Additional context
* It seems that it takes a while to process all data after the Toast message appears when you have a bigger history (mine has a little above 18,300 valid videos and around 4,000 ads)
* I filter out non-YouTube items (for sanity), as well as searches and Google Ads views
